Zend/zend_builtin_functions.stub.php

```diff
@@ -48,10 +48,10 @@ function get_mangled_object_vars(object $object): array {}
 function get_class_methods(object|string $object_or_class): array {}
 
 /** @param object|string $object_or_class */
-function method_exists(mixed $object_or_class, string $method): bool {}
+function method_exists($object_or_class, string $method): bool {}
```

Comment:
Can't it be
```suggestion
function method_exists(object|string $object_or_class, string $method): bool {}
```
?

Reply:
The problem is that the behavior is not compatible with weak types, because `method_exists("1", "foo")` silently return false, while `method_exists(1, "foo")` throws.

What we should have done here is to make both cases throw in PHP 8 (i.e. make it a requirement that the first argument is an existing class). However, it's too late for that now.